src/ast.mli

```diff
@@ -0,0 +1,91 @@
+type program = None
+
+type bin_op =
+  Add    | Minus | Multiply | Divide
+  | Modulo | Equal | Neq      | Lt
+  | Leq    | Gt    | Geq      | And
+  | Or
+
+type u_op = Not | Neg
+
+type p_type = Int_t | Bool_t | Double_t | Char_t | Unit_t
+
+type cont_type = String_t | List_t | Set_t | Map_t | Tuple_t
+
+type actor_op =
+  | Actor_send
+  | Actor_broadcast
+  | Actor_receive
+
+type types =
```

Comment:
'none<int>' is not possible with this definition as the None_param is just a placeholder for the lack of a parameter following a 'none'. So when an optional is a 'none', the optional_type will be None_t (which represents the 'none' part) and the optional_param will be None_param (which represents the "nothing", or empty string, that follows 'none').